src/backend/distributed/connection/connection_management.c

```diff
@@ -469,6 +485,31 @@ FinishConnectionEstablishment(MultiConnection *connection)
 			 * poll() after signal arrival.
 			 */
 			pollResult = poll(&pollFileDescriptor, 1, checkIntervalMS);
+#else /* !HAVE_POLL */
+			fd_set readFileDescriptorSet;
+			fd_set writeFileDescriptorSet;
+			fd_set exceptionFileDescriptorSet;
+			int selectTimeoutUS = checkIntervalMS * 1000;
+			struct timeval selectTimeout = { 0, selectTimeoutUS };
+			int selectFileDescriptor = PQsocket(connection->pgConn);
+
+			FD_ZERO(&readFileDescriptorSet);
+			FD_ZERO(&writeFileDescriptorSet);
+			FD_ZERO(&exceptionFileDescriptorSet);
+
+			if (pollmode == PGRES_POLLING_READING)
+			{
+				FD_SET(selectFileDescriptor, &readFileDescriptorSet);
+			}
+			else if (pollmode == PGRES_POLLING_WRITING)
+			{
+				FD_SET(selectFileDescriptor, &writeFileDescriptorSet);
+			}
+
+			pollResult = (select) (selectFileDescriptor + 1, &readFileDescriptorSet,
```

Comment:
why are the parenthesis around `select`?

Reply:
This might not be necessary since I think `FRONTEND` isn't defined while citus builds, but it was to make sure we don't hit [the `select` macro](https://github.com/postgres/postgres/blob/REL_10_STABLE/src/include/port/win32.h?utf8=%E2%9C%93#L374). We really just want to do our own thing and not think about whatever layer postgres tries to put between us and the syscall.